SonarQube.Common/ProcessRunnerArguments.cs

```diff
@@ -127,18 +127,17 @@ public static bool ContainsSensitiveData(string text)
             return SensitivePropertyKeys.Any(marker => text.IndexOf(marker, StringComparison.OrdinalIgnoreCase) > -1);
         }
 
-        public static string GetQuotedArg(string arg)
+        /// <summary>
+        /// According to https://msdn.microsoft.com/en-us/library/system.diagnostics.processstartinfo.arguments(v=vs.110).aspx
+        /// quotes have to be doubled. Also, to avoid problems when the argument has spaces or chars such as & which are problematic in .bat files, 
+        /// enclose the entire argument in quotes
+        /// </summary>
+        private static string EscapeArgument(string arg)
         {
             Debug.Assert(arg != null, "Not expecting an argument to be null");
-
-            string quotedArg = arg;
-
-            // If an argument contains a quote then we assume it has been correctly quoted.
```

Comment:
The previous assumption - i.e. do nothing if quotes are already present is invalid - we need to be able to handle args such as a" b "a & b" a"&b etc.